// craigslist-searcher/distribution/index.js
'use strict';

var _axios = require('axios');

var _axios2 = _interopRequireDefault(_axios);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var regexFindResult = /<p class="result-info">.+?datetime="(.+?)".+?href="(.+?)".+?>(.+?)<\/a>.+?price">(.+?)<.+?hood">\s*?\((.+?)\)<.+?<\/p>/g;
var baseSearchUrl = '.craigslist.org/search/';

var search = function search() {
  var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
      city = _ref.city,
      query = _ref.query,
      category = _ref.category,
      _ref$offset = _ref.offset,
      offset = _ref$offset === undefined ? 0 : _ref$offset,
      _ref$sort = _ref.sort,
      sort = _ref$sort === undefined ? 'rel' : _ref$sort;

  return new Promise(function (resolve, reject) {
    if (!city) reject(new Error('city is a required paramter.'));
    _axios2.default.get('https://' + city + baseSearchUrl + (category || ''), {
      params: {
        query: query ? query.replace(/\s/g, '+') : query,
        s: offset,
        sort: sort
      }
    }).then(function (_ref2) {
      var data = _ref2.data;

      var formatedData = data.replace(/\n/g, '');
      var dataResult = [];
      var matchArray = void 0;
      while (matchArray = regexFindResult.exec(formatedData)) {
        dataResult.push({
          datetime: matchArray[1],
          url: matchArray[2],
          title: matchArray[3],
          price: matchArray[4],
          region: matchArray[5]
        });
      }
      resolve(dataResult);
    }).catch(function (error) {
      return console.warn(error);
    });
  });
};

// export const detail = url => {};

module.exports = {
  search: search
};
